scripts/ibutsu-pod.sh

```diff
@@ -1,7 +1,77 @@
 #!/bin/bash
+
+# Some variables
 POD_NAME="ibutsu"
 JWT_SECRET=$(cat /dev/urandom | env LC_CTYPE=C tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1)
+IS_PERSISTENT=false
+CREATE_ADMIN=false
+CREATE_PROJECT=false
+POSTGRES_EXTRA_ARGS=
+REDIS_EXTRA_ARGS=
+BACKEND_EXTRA_ARGS=
+
+function print_usage() {
+    echo "Usage: ibutsu-pod.sh [-h|--help] [-p|--persistent] [-A|--create-admin] [-P|--create-project] [POD_NAME]"
+    echo ""
+    echo "optional arguments:"
+    echo "  -h, --help            show this help message"
+    echo "  -p, --persistent      persist the data in the containers"
+    echo "  -A, --create-admin    create an administrator ('admin@example.com')"
+    echo "  -P, --create-project  create a default project ('my-project')"
+    echo "  POD_NAME              the name of the pod, 'ibutsu' if ommitted"
+    echo ""
+}
+
+# Parse the arguments
+for ARG in $*; do
+    if [[ "$ARG" == "-p" ]] || [[ "$ARG" == "--persistent" ]]; then
+        IS_PERSISTENT=true
+    elif [[ "$ARG" == "-A" ]] || [[ "$ARG" == "--create-admin" ]]; then
+        CREATE_ADMIN=true
+    elif [[ "$ARG" == "-P" ]] || [[ "$ARG" == "--create-project" ]]; then
+        CREATE_PROJECT=true
+    elif [[ "$ARG" == "-h" ]] || [[ "$ARG" == "--help" ]]; then
+        print_usage
+        exit 0
+    else
+        POD_NAME=$ARG
+    fi
+done
+
+# Check this is being run in the correct directory
+DIRS=`ls -p | grep '/'`
+if [[ ! ${DIRS[*]} =~ "backend" ]] && [[ ! ${DIRS[*]} =~ "frontend" ]]; then
+    echo "This script needs to be run from the root of the project"
+    exit 1
+fi
 
+# If the data is persistent, check if there's a JWT secret in a file, or save it
+if [[ $IS_PERSISTENT = true ]]; then
+    if [[ ! -f ".jwtsecret" ]]; then
+        echo $JWT_SECRET > .jwtsecret
+    else
+        JWT_SECRET=$(<.jwtsecret)
+    fi
+fi
+
+# Persist the data
+if [[ $IS_PERSISTENT = true ]]; then
+    POSTGRES_EXTRA_ARGS+=' -v ./.postgres-data:/var/lib/postgres/data:Z'
+    REDIS_EXTRA_ARGS+=' -v ./.redis-data:/data:Z'
+    if [[ ! -d ".postgres-data" ]]; then
+        mkdir .postgres-data
+    fi
+    if [[ ! -d ".redis-data" ]]; then
+        mkdir .redis-data
+    fi
```

Comment:
Could also make these volumes with the `podman volume create` command and use them in vars above.
```bash
if volumes don't exist:
    podman volume create postgres-data
    podman volume create redis-data
POSTGRES_EXTRA_ARGS+=' -v postgres-data:/var/lib/postgres/data:Z'
REDIS_EXTRA_ARGS+=' -v redis-data:/data:Z'
```
Sorry I've never put in the time to understand bash if statements....

Reply:
I've never been a huge fan of volumes. They're a bit opaque, and can sometimes cause issues when you don't realise that there's a volume with the same name somewhere else.

Reply:
I added an option to use volumes.